### ISSUE TITLE
std::optional wrapper breaks MSYS2 builds 

### ISSUE THREAD

**Issue by elelel** [jtv/libpqxx#6]
Commit d6f3057e2f66006b8d728069aca7b1a939e19665 breaks MSYS2 builds on Windows (mingw64, i686, gcc version 6.2.0) :

```
In file included from ../include/pqxx/field:19:0,
                 from ../include/pqxx/result.hxx:31,
                 from ../include/pqxx/result:19,
                 from ../include/pqxx/binarystring.hxx:26,
                 from ../include/pqxx/binarystring:18,
                 from connection_base.cxx:46:
../include/pqxx/field.hxx: In member function 'int pqxx::field::get_opt() const':
../include/pqxx/field.hxx:215:36: error: expected primary-expression before ')' token
     if (is_null()) return OPTIONAL();
                                    ^
```

**Comment by jtv** [comment]
Hmm...  Can't figure out why that should happen!  The same compiler version works for me on Linux.

Is there any more information you can get from your compiler?  It looks to me as if its implementation of "optional" doesn't have a default constructor.  Could your installation be picking the wrong headers?

**Comment by elelel** [comment]
The problem is with the identifier chosen for the template param. In general, I think it's better to avoid all-caps single-word identifiers as they have a long history of colliding with C macros. I've created a pull request (10) with a fix.

**Comment by elelel** [comment]
I've just browsed the pull requests for this repo. I think this is the same error they're trying to fix in pull request #9 by cutting out the whole thing with #ifdefs. So probably pull request #10 will fix the problem under MSVC compilers as well.

**Comment by jtv** [comment]
Yup.  Thanks for the sleuthing!
